File: ios/VideoPlayerViewManager.m

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(VideoPlayerViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(uri, NSString)
RCT_EXPORT_VIEW_PROPERTY(paused, BOOL)
RCT_EXPORT_VIEW_PROPERTY(muted, BOOL)
RCT_EXPORT_VIEW_PROPERTY(volume, float)
RCT_EXPORT_VIEW_PROPERTY(resizeMode, NSString)
@end

@interface RCT_EXTERN_MODULE(NativeVideoPlayer, NSObject)
RCT_EXTERN_METHOD(getConstants)
@end

File: ios/VideoPlayerViewManager.swift
// VideoPlayerViewManager.swift
import Foundation

@objc(VideoPlayerViewManager)
class VideoPlayerViewManager: RCTViewManager {
    override func view() -> UIView {
        return VideoPlayerView()
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
}

@objc(NativeVideoPlayer)
class NativeVideoPlayer: NSObject {
    @objc
    static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    @objc
    func getConstants() -> [String: Any] {
        return [
            "RESIZE_MODE_CONTAIN": "contain",
            "RESIZE_MODE_COVER": "cover",
            "RESIZE_MODE_STRETCH": "stretch"
        ]
    }
}
